src/platform/posix/tcp_ip_channel.c

```diff
@@ -89,11 +154,53 @@ static lf_ret_t TcpIpChannel_try_connect(NetworkChannel *untyped_self) {
 
     int ret = connect(self->fd, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
     if (ret < 0) {
-      return LF_COULD_NOT_CONNECT;
+      if (errno == EINPROGRESS) {
+        self->client_connect_in_progress = true;
+        LF_DEBUG(NET, "Connection in progress!");
+        return LF_TRY_AGAIN;
+      } else {
+        LF_ERR(NET, "Connect failed errno=%d", errno);
+        self->client_connect_in_progress = false;
+        TcpIpChannel_reset_socket(self);
+        return LF_TRY_AGAIN;
```

Comment:
At first it was a little bit confusing for me that in progress and failed both return `TRY_AGAIN`, but I guess it does somewhat make sense from the `NetworkChannel` perspective.
In both cases we need to try again.

But I see one argument why maybe a differentiation could make sense here.

In the Zephyr example you sleep for some time if you get a `TRY_AGAIN`, but this only makes sense for the `EINPROGRESS` case from what I can see.

In case of a `reset_socket` we could just immediately try again.

So maybe return `LF_IN_PROGRESS` For `EINPROGRESS` and only wait for that case?

Reply:
And then also maybe we could be more transparent and instead of `LF_TRY_AGAIN` return a `LF_CONNECTION_FAILED`? The user of the API can then decide if they want to try again or not I guess?
But I am also okay with keeping `TRY_AGAIN`. It has the advantage of telling the user what to do ;)

Reply:
Another way would be `LF_IN_PROGRESS_TRY_AGAIN` and `LF_FAILED_TRY_AGAIN`

Reply:
I went for `LF_IN_PROGRESS` and `LF_TRY_AGAIN`, on Zephyr we do a short sleep before trying again.